vega/vegalite.py

```diff
@@ -18,36 +20,55 @@
 }
 
 
-def view(dataframe, spec={}):
-    """Create and immediately display even if it is not the last line."""
-    display(create(dataframe, spec))
-
-
-def create(dataframe, spec={}):
-    """Create vega-lite from a dataframe."""
-    return VegaLite(dataframe.columns, dataframe.values, spec)
+def view(spec, data=None):
+    """View a Vega-Lite visualization in the Jupyter notebook.
+    
+    Parameters
+    ----------
+    spec : dict
+        The Vega-Lite spec as a Python dict.
+    data : pandas.DataFrame
+        If the spec doesn't encode the data, it can be passed in separately.
+    """
+    spec = utils.prepare_spec(spec, data)
+    display(VegaLite(spec))
+
+
+def create(spec, data=None):
+    """Create Vega-Lite visualization from a spec and (optionaly) data.
+    
+    Parameters
+    ----------
+    spec : dict
+        The Vega-Lite spec as a Python dict.
+    data : pandas.DataFrame
+        If the spec doesn't encode the data, it can be passed in separately.
+    """
+    return VegaLite(spec)
 
 
 class VegaLite(object):
-    """Define Vega-Lite widget."""
-
-    def __init__(self, columns, data, spec):
-        """Initialize Vega-Lite widget.
-
-        Pass in a list of columns, and data values.
-        """
-        self.columns = columns
-
-        updated = utils.update(DEFAULTS, spec)
-
-        self.spec = utils.update({
-            "data": {
-                "values": utils.data(data, columns)
-            }
-        }, updated)
-
-    def _repr_html_(self):
-        """Used by the frontend to show html."""
-        template = utils.get_content(TEMPLATE)
-
-        return template.format(spec=json.dumps(self.spec))
+    """A custom Vega-Lite display object."""
+
+    def __init__(self, spec):
+        self.spec = utils.update(spec, DEFAULTS, overwrite=False)
```

Comment:
I think putting ``DEFAULTS`` in the first argument will end up modifying ``DEFAULTS``, so it will work if run once, but not work correctly the second time.